pagopa/io-app
# fix: [IOAPPCOM-16] Message spinner from foreground push notification

## Short description
This PR fixes the loading spinner not showing when automatically refreshing the message list, after a foreground push notification has been received and selected.

## List of changes proposed in this pull request
- Boolean conditions have been refactored on the `MessageList.ts` component in order to properly map all cases.
- Condition `messages.length > 0` has been shared in multiple places using a `const`
Please refer to the following table to understand the boolean conditions (keep in mind that every time the user initiates a pull-down-to-refresh, that is a case where the top spinner has to be shown).
![Screenshot 2022-12-16 at 17 48 53](https://user-images.githubusercontent.com/5150343/208148181-821f9c5c-69b5-430c-8c6c-00607c01d18c.png)

## How to test
There are multiple tests that need to be done:
- starting the application when stopped/killed should show the footer spinner
- refreshing the list by pull-down-to-refresh should show the top spinner
- refreshing the list from a foreground push notification, while on the message list, should show the top spinner
- auto-refresh of messages (after the one minute timeout) should show the top spinner
- when the end of the list is reached and there are more messages to show, the footer spinner should be shown

For each of these cases, make sure to test both a successful and a failing response from the message API (you can use a tool like [Proxyman](https://proxyman.io/) to intercept the request and making it fail. Alternatively, edit the responses from the `io-dev-api-server`) and also to repeat the tests for the Archived messages (apart for the foreground push notification, which does not refresh the archive messages).

Use the dev-server configuration with more than 12 messages to trigger the "load more messages" at the end of the list scrolling
